qemu/tests/rng_hotplug.py

```diff
@@ -111,15 +126,26 @@ def stop_rngd(vm):
             hotplug_rng(vm, new_dev)
             dev_list.append(new_dev)
 
-        # Run test after hotplug
-        if test_after_hotplug and i == xrange(repeat_times)[-1]:
-            restart_rngd(vm)
-            error_context.context("Run %s after hotplug" % test_after_hotplug,
-                                  logging.info)
-            utils_test.run_virt_sub_test(test, params, env,
-                                         test_after_hotplug)
+        # Run test after the last hotplug
+        if i == repeat_times - 1:
+            #run rng test after hotplug
+            check_and_run_subtest(test_after_hotplug)
+
+            #run reboot/shutdown after hotplug
+            check_and_run_subtest(pm_test_after_plug)
+
+            if vm.is_dead():
```

Comment:
If vm.is_dead(), would the test be passed or failed?  Do we need to check the sub test here? IMO, if the test is shutdown, it will be expected. Otherwise, it should fail.  And better to wait seconds before check if vm is dead.

Reply:
1.If vm is dead,job will pass.
   In my understanding,if the sub test failed,job will fail at check_and_run_subtest() function and not go to is_dead() function at all,so it will not affect the result.
   Please correct me if I misunderstand it,thanks

2.About wait seconds to check vm status, subtest shutdown will wait untill guest is down as described as following,so I think no need to wait again.   
   shutdown.py
    KVM shutdown test:
    1) Log into a guest
    2) Send a shutdown command to the guest, or issue a system_powerdown
       monitor command (depending on the value of shutdown_method)
    3) Wait until the guest is down